### ISSUE TITLE
Replace dict.has_key() with the 'in' keyword, for twisted/conch

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8362]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8362|
|Type|enhancement|
|Created|2016-05-26 17:42:47Z|

In PEP 0290, has_key() was deprecated in Python 2.2 and higher: 
* https://www.python.org/dev/peps/pep-0290/#testing-dictionary-membership
* https://docs.python.org/2.2/whatsnew/node4.html

In Python 3, dict.has_key() was removed:
* https://docs.python.org/3.0/whatsnew/3.0.html#builtins

PEP 0290 says that for testing dictionary membership, this:

```
if d.has_key(k):
```

should be changed to this:

```
if k in d:
```

<details><summary>Searchable metadata</summary>

```
trac-id__8362 8362
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__None None
time__1464284567067308 1464284567067308
changetime__1465027382322196 1465027382322196
version__None None
owner__glyph glyph

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

https://github.com/twisted/twisted/pull/78

This was pulled out from [ticket:4053]

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set owner to @glyph|
|-|-|

Looks pretty unambiguously correct.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph set status to `closed`|
|-|-|

In changeset b0581ace1bbdbbe27ca3493d11124c0e05ca9d62
```
#!CommitTicketReference repository="" revision="b0581ace1bbdbbe27ca3493d11124c0e05ca9d62"
Merge pull request #78 from rodrigc/has_key4

Author: rodrigc

Reviewer: glyph

Fixes: #8362

Remove uses of deprecated `has_key` dict method from twisted.conch.
```
